url8f84d8d956_klapaukh_JTuttePoly_tgz-pJ8-BitTutteJ8/src/BitTutte.java

```diff
@@ -0,0 +1,513 @@
+	 * @return Graph g with all the vertices added.
+	 */
+	private Graph orderVertices(Graph g, List<Pair<Integer, Integer>> l, Map<Integer, Integer> numbers, int vertexHeuristic) {
+		// TODO UNTESTED
+		List<Pair<Integer, Integer>> counts = new ArrayList<Pair<Integer, Integer>>();
+		for (Integer i : numbers.keySet()) {
+			counts.add(new Pair<Integer, Integer>(i, 0));
+		}
+		Set<Pair<Integer, Integer>> underlyingEdges;
+		switch (vertexHeuristic) {
+		case V_RANDOM:
+			for (int i = 0; i < counts.size(); i++) {
+				counts.set(i, new Pair<Integer, Integer>(counts.get(i).first(), (int) (Math.random() * 2000)));
+			}
+			Collections.sort(counts, new Pair.MaximizeSecondComparator());
+			break;
+		case V_MINIMISE_UNDERLYING_DEGREE:
+			underlyingEdges = new TreeSet<Pair<Integer, Integer>>(l);
+			for (Pair<Integer, Integer> i : underlyingEdges) {
+				counts.set(numbers.get(i.first()), new Pair<Integer, Integer>(i.first(), counts.get(numbers.get(i.first())).second() + 1));
+				counts.set(numbers.get(i.second()), new Pair<Integer, Integer>(i.second(), counts.get(numbers.get(i.second())).second() + 1));
+			}
+			Collections.sort(counts, new Pair.MinimizeSecondComparator());
+			break;
+		case V_MAXIMISE_UNDERLYING_DEGREE:
+			underlyingEdges = new TreeSet<Pair<Integer, Integer>>(l);
+			for (Pair<Integer, Integer> i : underlyingEdges) {
+				counts.set(numbers.get(i.first()), new Pair<Integer, Integer>(i.first(), counts.get(numbers.get(i.first())).second() + 1));
+				counts.set(numbers.get(i.second()), new Pair<Integer, Integer>(i.second(), counts.get(numbers.get(i.second())).second() + 1));
+			}
+			Collections.sort(counts, new Pair.MaximizeSecondComparator());
+			break;
+		case V_MINIMISE_DEGREE:
+			for (Pair<Integer, Integer> i : l) {
+				counts.set(numbers.get(i.first()), new Pair<Integer, Integer>(i.first(), counts.get(numbers.get(i.first())).second() + 1));
+				counts.set(numbers.get(i.second()), new Pair<Integer, Integer>(i.second(), counts.get(numbers.get(i.second())).second() + 1));
+			}
+			Collections.sort(counts, new Pair.MinimizeSecondComparator());
+			break;
+		case V_MAXIMISE_DEGREE:
+			for (Pair<Integer, Integer> i : l) {
+				counts.set(numbers.get(i.first()), new Pair<Integer, Integer>(i.first(), counts.get(numbers.get(i.first())).second() + 1));
+				counts.set(numbers.get(i.second()), new Pair<Integer, Integer>(i.second(), counts.get(numbers.get(i.second())).second() + 1));
+			}
+			Collections.sort(counts, new Pair.MaximizeSecondComparator());
+			break;
+		case V_BFS:
+			System.out.println("BFS vertex order is unimplemented, falling back to none");
+		case V_NONE:
+			// just do nothing
+		default:
+			break;
+		}
+
+		for (int i = 0; i < counts.size(); i++) {
+			numbers.put(counts.get(i).first(), i);
+		}
+
+		for (Pair<Integer, Integer> i : l) {
+			g.addEdge(numbers.get(i.first()), numbers.get(i.second()));
+		}
+
+		return g;
+	}
+
+	private FactorPoly tutte(Graph graph, int mid) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 84 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.